.github/workflows/action-test-development.yml

```diff
@@ -0,0 +1,60 @@
+# Action that runs the full development stack:
+# Steps (Python-only):
+# - creates esmvaltool conda environment and pip-installs esmvaltool
+# - downloads latest master of esmvalcore and installs it in development mode
+# - runs tests of esmvaltool
+# Triggered by a push to master and nightly
+
+name: Develop Test
+
+# runs on a push on master and at the end of every day
+on:
+  push:
+    branches:
+    - master
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  linux:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+      fail-fast: false
+    name: Linux Python ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: esmvaltool
+          environment-file: environment.yml
+          python-version: ${{ matrix.python-version }}
+          miniconda-version: "latest"
+          channels: conda-forge
+      - shell: bash -l {0}
+        run: mkdir -p develop_test_linux_artifacts_python_${{ matrix.python-version }}
+      - shell: bash -l {0}
+        run: conda --version 2>&1 | tee develop_test_linux_artifacts_python_${{ matrix.python-version }}/conda_version.txt
+      - shell: bash -l {0}
+        run: python -V 2>&1 | tee develop_test_linux_artifacts_python_${{ matrix.python-version }}/python_version.txt
```

Comment:
It might be nicer to use a `conda env export` here instead, that would get you all the required version info to reproduce the environment locally, especially if you do it after the pip install command.

Reply:
there is one right under it :grin: